core/src/main/java/io/questdb/MessageBusImpl.java

```diff
@@ -446,37 +446,20 @@ public MCSequence getWalTxnNotificationSubSequence() {
 
     @TestOnly
     public void clear() {
-        columnPurgePubSeq.clear();
```

Comment:
We could keep this method as is if we only call it in the `tearDown` method.

Reply:
for "clean" bus it is sufficient to have both pub- and sub-sequences to be in "empty" queue state. Their seqeunce numbers need not to be at their initial values, there is very little upside in trying to achieve that.

The sub- sequence is "cleared" by consuming the entire sequence until the queue is empty. This is thread-safe method of emptying the queue. I don't see equivalent method of  moving pub- sequence in the safe spot. Resetting pub- sequence will also require sub-sequence to be set to 0 and there is no thread-safe way of doing that either.

Hope this makes sense?

Reply:
sure, i understood this! I am just not keen on adding a fragile .clear() method to pub- sequence. Chances are it will be a source of friction in the future.

Reply:
Maybe we could mark it as `@TestOnly` and call it something like `clearUnsafe`?